extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/providers/DsChannelTypeProvider.java

```diff
@@ -47,6 +51,7 @@
  * @author Matthias Siegele - Initial contribution
  *
  */
+@Component(service = ChannelTypeProvider.class, immediate = true)
```

Comment:
The base class has an `activate()` method. Therefore this should override in in order to make it explicit:

```java
@Activate
@Override
public void activate(ComponentContext componentContext) {
    super.activate(componentContext);
}
```

The `deactivate()` method correspondingly.

Note: it correctly works as it is because the SCR implementations will look for such method signatures automatically. However, it is considered good practice to make it explicit - and therefore they need to be overridden because the XML generation processor does _not_ look at superclasses.